src/Data/List/Relation/Unary/Any.agda

```diff
@@ -66,6 +66,10 @@ module _ {P : Pred A p} where
   lookup : ∀ {xs} → Any P xs → A
   lookup {xs} p = List.lookup xs (index p)
 
+  lookup-p : ∀ {xs} → (p : Any P xs) → P (lookup p)
```

Comment:
I think this belongs in `Any.Properties` and how about calling it something slightly more meaningful (though I agree, it's a seriously hard one to name!). How about `lookup-result`?